### DISCUSSION TITLE
Ambiq Suite SDK vs. Mbed OS vs. Arduino Cores V1.x/2.x

### DISCUSSION THREAD

**Discussion by idea--list** [sparkfun/Arduino_Apollo3#357] [General]
@oclyke @Wenn0101 
Could you please summarize how all your development is related to all these different platforms?

I mean i regularly read all the news/blogposts on sparkfun.com ever since the Apollo boards have been announced but i am still not 100% sure if i have a proper understanding or not.

For sure the Ambiq Suite SDK is the base for all other implementations. Meaning we can also include raw Ambiq SDK code in our Arduino or Mbed OS projects (at least with Mbed OS it seems to be that way). If i am right most of the work was based on Ambiq Suite SDK V2.4.2.

I also see there are a bunch of different branches based on Ambiq Suite SDK V2.5.1 over [here](https://github.com/sparkfun/AmbiqSuiteSDK), however it seems as if things got stuck with V2.5.1 as there are some issues. Or maybe even Ambiq might release a newer SDK version as the Apollo4 chips will get released soon and you are hoping the newer SDK will have less flaws?

If i understand it right Sparkfun began porting the Apollo3 chips to Mbed OS based on Ambiq SDK 2.4.2 and ultimately the resulting work has been merged into the official Mbed OS V6.3 release.

I also guess Arduino core V1.x is a plain regular Arduino core without involving any Mbed. Is that right?
I can see most of the work/bug fixes continue on the Arduino core V2.x line which is an Mbed OS/Arduino hybrid.
If that is true:

- Do you keep the Mbed part of Arduino core V2.x synchronized with the official Mbed OS releases?
- Can we please have a list of the different Arduino APIs which are implemented in plain Arduino and which are based on Mbed code?

I am asking all these because if you find a bug in the Arduino core V2.x then all the above info could guide the community if the bug might be also present in the current official Mbed OS release, in which case the community could port the fixes of the Arduino core much faster to Mbed OS in case that is also affected. Of course it would also facilitate porting of bug fixes from Mbed OS to the Arduino Core V2.x.

**Comment by Wenn0101** [comment]
@idea--list I think it has been a little bit too long since we had a blog post to update the wider community about Artemis. I will work on remedying that, but I did want to address some of your questions specifically here.

You are correct that the AmbiqSuite is the base for these projects and that you can use functionality from the ambiq suite in your Mbed and Arduino projects. Both of these projects use AmbiqSuite 2.4.2. You are also correct in sayin that we started work with AmbiqSuite 2.5.1. We ended up pulling a release at last second which included an update to AmbiqSuite 2.5.1, because it failed some of our pre-release testing. We have not prioritized working on this since, but you can expect a future update which includes this version of the AmbiqSuite.

You are also correct that the v1 core is a 'bare-metal' implementation and starting on v2, Mbed OS provides the core to our Arduino core.
This version of Mbed is actually a custom branch that we have of Mbed5. Even though the Artemis targets did not officially make it into Mbed until Mbed 6, development began before this. One of our motivations for switching to an Mbed core, was for easier integration into the ArduinoBLE library. Mbed6 saw some changes to the BLE API, that have not made their way to the ArduinoBLE library yet. For this reason we decided to have our Arduino core use a version of Mbed5, that we maintain [here](https://github.com/sparkfun/mbed-os-ambiq-apollo3/tree/ambiq-apollo3-arduino). Every release of the Arduino core includes a static library that is compiled from the latest commit on this branch. In the future, we will likely update this to run a more recent version of Mbed, but we may avoid tying ourselves to the latest release every branch so that we have less unexpected surprised before release.
I don't immediately have a list of which functions in the Arduino API directly call Mbed functionality, and which use a more direct approach, I will work on this list and post back here when I have of it.

**Comment by idea--list** [comment]
@Wenn0101 
Thanks for all the clarifications!

_For this reason we decided to have our Arduino core use a version of Mbed5, that we maintain here._
This is good to bear in mind and find it a great idea to avoid all the bugs that come from the bleeding-edge like nature of Mbed.
However it would be great to include this info at the top of the readme.md of that repo. That way it would be clear for everyone what is the purpose of that repo and how that relates to the official Mbed OS releases and the Arduino core.
